src/adapters/winston.js

```diff
@@ -0,0 +1,67 @@
+const assert = require('assert');
+
+/**
+  @class Winston
+  Adapter for the winston logger chain link.
+  Exposes logger's main functions as if a standard winston module was used
+  The list of functions:
+  @function info - log message with an info log level
+  @function warn - log message with an warn log level
+  @function error - log message with an error log level
+  @function debug - log message with a debug log level
+  @function silly - log message with a silly log level
+  @function verbose - log message with a verbose log level
+  @function log - log message with a provided log level
+  @function profile - profiling function
+  @constructor consumes the instance of a LoggerChain @class
+**/
+class Winston {
+  /**
+    @constructor
+    Construct the instance of a winston adapter
+    Create main logging functions
+    Each function has the following arguments:
+    @param chain {object} - an instance of a chain
+    @param chainLinkIndex {number} - index of a chain link this adapter is for
+  **/
+  constructor(chain, chainLinkIndex = -1) {
+    assert(chain);
+    this.chain = chain;
+    this.chainLinkIndex = chainLinkIndex;
+    const loggingLevels = ['error', 'warn', 'info', 'debug', 'silly', 'verbose'];
+    for (const loggingLevel of loggingLevels) {
+      this[loggingLevel] = (message, ...args) => {
```

Comment:
I am not sure I understand why, you just need to define it even as const a few lines up outside the for loop